streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImpl.java

```diff
@@ -196,10 +198,12 @@
                 retentionPeriod
             );
         }
-        final StoreBuilder<SessionStore<K, VR>> builder = Stores.sessionStoreBuilder(
+        final StoreBuilder<SessionStore<K, VR>> builder = new SessionStoreBuilder<>(
             supplier,
             materialized.keySerde(),
-            materialized.valueSerde()
+            materialized.valueSerde(),
+            Time.SYSTEM,
+            true
```

Comment:
Replacing `Stores.sessionStoreBuilder` with a call to `new SessionStoreBuilder` to be able to set this newly introduced flag to `true`. It's not super clean, but I did not have a better idea to tackle this (more details below).

Reply:
It's not exposed. `Stores` does set the flag hardcoded to `false` and `SessionStoreBuilder` is not part of public API. Still, it seems not clean for internal code. It basically leaks DSL into PAPI, but stores should be DSL agnostic...

Reply:
I thinks @mjsax 's original comment is regarding about passing this flag to the store builder, and hence to the caching store, whether it is the most elegant way.

I'm actually wondering if it is really necessary -- see my other comment below.